src/core.jl

```diff
@@ -134,6 +134,54 @@ Note: An exception will be raised if the edge is not in the graph.
 """
 rem_edge!(g::SimpleGraph, src::Int, dst::Int) = rem_edge!(g, Edge(src,dst))
 
+"""Remove the vertex `v` from graph `g`.
+This operation has to be performed carefully if one keeps external data structures indexed by
+edges or vertices in the graph, since internally the removal is performed swapping the vertices `v`  and `n=nv(g)`,
+and removing the vertex `n` from the graph.
+After removal the vertices in the ` g` will be indexed by 1:n-1.
+This is an O(k) operation, where `k` is the max of the degree of vertices `v` and `n`.
+Note: An exception will be raised if the vertex `v`  is not in the `g`.
+"""
+function rem_vertex!(g::SimpleGraph, v::Int)
+    v in vertices(g) || throw(BoundsError())
+    n = nv(g)
+
+    edgs = in_edges(g, v)
+    for e in edgs
+        unsafe_rem_edge!(g, e)
+    end
+    neigs = copy(in_neighbors(g, n))
+    for i in neigs
+        unsafe_rem_edge!(g, Edge(i, n))
+    end
+    if v != n
+        for i in neigs
+            unsafe_add_edge!(g, Edge(i, v))
+        end
+    end
+
+    if is_directed(g)
+        edgs = out_edges(g, v)
+        for e in edgs
+            unsafe_rem_edge!(g, e)
+        end
+        neigs = copy(out_neighbors(g, n))
+        for i in neigs
+            unsafe_rem_edge!(g, Edge(n, i))
+        end
+        if v != n
+            for i in neigs
+                unsafe_add_edge!(g, Edge(v, i))
+            end
+        end
+    end
+
+    g.vertices = 1:n-1
+    g.fadjlist = g.fadjlist[1:n-1]
+    g.badjlist = g.badjlist[1:n-1]
```

Comment:
OK, but will the removed data not be garbage collected since we're essentially making a view?